src/vec.rs

```diff
@@ -80,6 +82,18 @@ impl V3 {
     }
 }
 
+macro_rules! mul_by_matrix {
+    ($vec:expr,
+    $a00:expr, $a01:expr, $a02:expr,
+    $a10:expr, $a11:expr, $a12:expr,
+    $a20:expr, $a21:expr, $a22:expr
+    ) => (V3{
+        x: ($vec.x) * ($a00) + ($vec.y) * ($a01) + ($vec.z) * ($a02),
+        y: ($vec.x) * ($a10) + ($vec.y) * ($a11) + ($vec.z) * ($a12),
+        z: ($vec.x) * ($a20) + ($vec.y) * ($a21) + ($vec.z) * ($a22),
+    })
+}
```

Comment:
This can be redone into plain function. In case you wish keep this macro, remember that you don't have to wrap every fragment into parens, rustc will correctly expand it anyway.